server/bleep/src/webserver/studio.rs

```diff
@@ -680,6 +681,62 @@ async fn generate_llm_context(app: Application, context: Vec<ContextFile>) -> Re
     Ok(s)
 }
 
+/// If a given studio's name is `NULL`, try to auto-generate a name.
+///
+/// If the requested studio already has a name, this is a no-op.
+async fn populate_studio_name(
+    app: Extension<Application>,
+    user: Extension<User>,
+    studio_id: i64,
+) -> webserver::Result<()> {
+    let user_id = user.login().ok_or_else(no_user_id)?.to_string();
+
+    let snapshot_id = latest_snapshot_id(studio_id, &*app.sql, &user_id).await?;
+    let needs_name = sqlx::query! {
+        "SELECT id FROM studios WHERE id = ? AND name IS NULL",
+        studio_id,
+    }
+    .fetch_optional(&*app.sql)
+    .await?
+    .is_some();
+
+    if !needs_name {
+        return Ok(());
+    }
+
+    let (context_json, messages_json) = sqlx::query! {
+        "SELECT context, messages FROM studio_snapshots WHERE id = ?",
+        snapshot_id,
+    }
+    .fetch_one(&*app.sql)
+    .await
+    .map(|r| (r.context, r.messages))?;
+
+    let llm_gateway = app
+        .llm_gateway_client()
+        .map_err(|e| Error::user(e).with_status(StatusCode::UNAUTHORIZED))?
+        .model(LLM_GATEWAY_MODEL)
+        .temperature(0.0);
+
+    let messages = &[llm_gateway::api::Message::system(
+        &prompts::studio_name_prompt(&context_json, &messages_json),
+    )];
+
+    let name = llm_gateway
+        .chat(messages, None)
```

Comment:
can we use 3.5-turbo heres/

Reply:
Have tested and GPT-3.5 turbo seems sufficient